src/vm/wren_core.c

```diff
@@ -13,6 +13,26 @@
 
 #include "wren_core.wren.inc"
 
+#ifdef __DJGPP__
+double fmax(double x, double y)
+{
+  if (isnan(x))
```

Comment:
- Why would double not be a number?
- What is y isn't a number either?
- If one or both aren't "numbers" what would just falling thru to the comparison do?

Reply:
1/ not all 2^64 values possibles for a double are numbers. Like `sqrt(-1)`.
2/ if both are NaN, you can only return NaN.
3/ all binary operations (except `==`) returns `false` if any (or both) is NaN, therefore `x` would be returned. But then, the second `if` isn't strictly necessary.

Reply:
You can check musl for a standard lib implementation https://git.musl-libc.org/cgit/musl/tree/src/math/fmax.c

Reply:
> * Why would double not be a number?
e.g. 0.0/0.0, infinity * 0.0, and so on
> * What if y isn't a number either?
By the doc you need to return NaN
> * If one or both aren't "numbers" what would just falling thru to the comparison do?
It's possible to optimize a bit by removing second check but code becomes way more unintuitive and error-prone. Second one is strictly necessary to be correct in case of x=nan, y=1 which needs to return 1.

According to the fmin/fmax doc:
RETURN VALUE
       These functions return the minimum [maximum] of x and y.

       If one argument is a NaN, the other argument is returned.

       If both arguments are NaN, a NaN is returned.

Reply:
> You can check musl for a standard lib implementation https://git.musl-libc.org/cgit/musl/tree/src/math/fmax.c

Thanks. I'll update based on it

Reply:
Updated